kvbc/src/v4blockchain/v4_blockchain.cpp

```diff
@@ -105,6 +105,7 @@ BlockId KeyValueBlockchain::add(categorization::Updates &&updates) {
 BlockId KeyValueBlockchain::add(const categorization::Updates &updates,
                                 v4blockchain::detail::Block &block,
                                 storage::rocksdb::NativeWriteBatch &write_batch) {
+  getVerifiabilityManagementLayer().getBuilder()->ProcessUpdates(updates);
```

Comment:
Doesn't it need the block id as well to know the version?

Reply:
Our implementation of the MPT is not strictly coupled with the block height. Every set of keys/values added to a MPT bumps the version by 1.